packages/cloud_firestore_odm/cloud_firestore_odm_generator/lib/src/collection_data.dart

```diff
@@ -346,7 +346,7 @@ represents the content of the collection must be in the same file.
         type.isDartCoreInt ||
         type.isDartCoreDouble ||
         type.isDartCoreBool ||
-        type.isPrimitiveList ||
+        type.isPrimitiveIterable ||
```

Comment:
We don't want to allow any Iterable. Iterables may have an infinite length. 

We only want fixed-length collections. Such as List/Set